### ISSUE TITLE
secret-in-log: eval / bash -c / trap 文字列内の sink を解析する

### ISSUE THREAD

**Issue by ultra-supara** [ultra-supara/sisakulint#444]
## 概要

`secret-in-log` の sink 検出は `syntax.Walk` で CallExpr を拾う方式だが、以下のパターンでは sink 相当の処理がシェル AST 上 CallExpr として見えず、または別コンテキストで実行されるため検出できない:

1. **`eval '...'`** — 引数文字列は Lit としてのみ parse され、内部の `echo $TOKEN` は CallExpr にならない。
2. **`bash -c '...'`** / `sh -c '...'` / `zsh -c '...'` — 同様。`-c` の引数は Lit。
3. **`trap '…' ERR|EXIT|DEBUG`** — trap body は文字列として保存され、実行時に eval される。parse pass 時点では CallExpr ツリーに入らない。
4. **関数定義本体** — `foo() { echo $TOKEN; }; foo` の本体はシェル関数として FuncDecl.Body に入る。現状の `syntax.Walk` は FuncDecl 内を歩くので本体内の `echo` は拾われる想定だが、関数呼び出し時に tainted 引数が `$1` として入るケース（`foo "$TOKEN"` + `echo $1` in body）は追跡できない。

## 漏洩例（現ルールでは未検出）

```yaml
- run: |
    TOKEN=$(echo "$KEY" | jq -r '.token')
    eval 'echo "$TOKEN"'          # 未検出
    bash -c 'echo "$TOKEN"'       # 未検出
    trap 'echo "failed with $TOKEN"' ERR   # 未検出、実行されれば log へ
    leak() { echo "$1"; }
    leak "$TOKEN"                 # 関数経由、未検出（本体は echo $1）
```

### D-1: eval / bash -c / sh -c / zsh -c の引数再 parse

- CallExpr の先頭が `eval` または `bash`/`sh`/`zsh` で `-c` オプション付きの場合、引数 Lit の文字列を **再度 `syntax.NewParser` で parse** し、得られた AST を現在の tainted 集合で `findEchoLeaks` に通す。
- 文字列結合 (`eval "echo $X"`) で Lit が ParamExp を含む場合は Lit 部分のみ再 parse し、ParamExp の tainted 判定は親コンテキストの `tainted` を流用する。

### D-2: trap body の再 parse

- CallExpr `trap` の第 1 引数（body 文字列）を同様に再 parse して解析。第 2 引数以降の signal 指定 (`ERR`/`EXIT`/`DEBUG`/`SIGTERM` ...) は解析不要。

### D-3: 関数引数の taint 伝播（オプション）

- コスト高め。初版はスキップし、`foo() { ... }` 内の `$1` `$2` は untracked とするのが現実的。D-1/D-2 が片付いてから再評価。

## リスク / 誤検出

- `eval` / `bash -c` 引数が動的に組み立てられている場合（`eval "$CMD"` のように変数参照だけ）は再 parse できないので skip（現状と同じ挙動）。
- 再 parse 失敗（シンタックスエラー等）は全体を skip して現状踏襲。

## 関連

- 親ルール: `secret-in-log`
- 関連コード: `pkg/core/secretinlog.go` の `findEchoLeaks`
- 関連 doc: `docs/secretinlogrule.md` の "Patterns the rule does not analyze at all" `eval` / `bash -c` / `trap` 項

---
_Related: epic #445 (TaintTracker AST 化エピック)。AST 化された TaintTracker と scope 対応が完了すれば、本 issue の sink 判定精度も向上する見込み。_
